File: 2533-bitwise-xor-of-all-pairings/bitwise-xor-of-all-pairings.py
class Solution:
    def xorAllNums(self, nums1: List[int], nums2: List[int]) -> int:
        x1, x2 = 0, 0
        if len(nums1) & 1:
            x2 = reduce(lambda x, y: x ^ y, nums2)
        if len(nums2) & 1:
            x1 = reduce(lambda x, y: x ^ y, nums1)
        return x1 ^ x2

File: 2533-bitwise-xor-of-all-pairings/bitwise-xor-of-all-pairings.java
class Solution {
    public int xorAllNums(int[] nums1, int[] nums2) {
        int res = 0;
        if ((nums1.length & 1) == 1) 
            res = Arrays.stream(nums2).reduce(0, (x, y) -> x ^ y);
        if ((nums2.length & 1) == 1) 
            res = Arrays.stream(nums1).reduce(res, (x, y) -> x ^ y);
        return res;
    }
}